linfa-trees/src/decision_trees/algorithm.rs

```diff
@@ -648,20 +797,23 @@ mod tests {
 
         // we should only use feature index 8 here
         assert_eq!(&model.features(), &[8]);
-        assert_eq!(
-            &model.feature_importance(),
-            &[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0]
-        );
+
+        let ground_truth = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0];
+
+        for (imp, truth) in model.feature_importance().iter().zip(&ground_truth) {
+            assert_abs_diff_eq!(imp, truth);
+        }
 
         // check for perfect accuracy
         let cm = model.predict(dataset.records()).confusion_matrix(&dataset);
-        assert!(cm.accuracy() == 1.0);
+        // can't be more than one, but >= avoids clippy complaining about float exact comparison
+        assert!(cm.accuracy() >= 1.0);
```

Comment:
we should use `approx` most of the time, this can be expressed as `assert_abs_diff_eq!(.., 1.0, epsilon = 1e-15)` as well

Reply:
do you know the default precision without specifying `epsilon`?

Reply:
I found this in the approx repo:
```rust

pub trait AbsDiffEq<Rhs = Self>: PartialEq<Rhs>
where
    Rhs: ?Sized,
{
    ....
    /// The default tolerance to use when testing values that are close together.
    ///
    /// This is used when no `epsilon` value is supplied to the [`abs_diff_eq!`], [`relative_eq!`],
    /// or [`ulps_eq!`] macros.
    fn default_epsilon() -> Self::Epsilon;
    ....
}

...

macro_rules! impl_signed_abs_diff_eq {
    ($T:ident, $default_epsilon:expr) => {
        impl AbsDiffEq for $T {
            ....
            fn default_epsilon() -> $T {
                $default_epsilon
            }
            ....
        }
    };
}

....

impl_signed_abs_diff_eq!(f32, f32::EPSILON);
impl_signed_abs_diff_eq!(f64, f64::EPSILON);
```
So it uses the epsilon value of the floating point type